src/tools/metrics/recorder.rs

```diff
@@ -75,6 +57,42 @@ impl TestMetrics {
     }
 }
 
+pub fn initialize() -> Snapshotter {
```

Comment:
Is this used anywhere else but in the `Default` impl for `TestMetrics`?

Reply:
Not that I know of.

Reply:
I'd suggest making a `new` function on `TestMetrics` that contains this logic and then have the default impl call `new`.